aspects/aspects.go

```diff
@@ -236,6 +314,48 @@ func (p accessPattern) isWriteable() bool {
 	return p.access == readWrite || p.access == write
 }
 
+// pattern is an individual part of a dot-separated name or path pattern. It
+// can be a literal value of a placeholder delineated by curly brackets.
+type pattern string
+
+func (p pattern) isPlaceholder() bool {
+	return p[0] == '{' && p[len(p)-1] == '}'
+}
```

Comment:
my preference is still in a follow-up to have interfaces and do this distinction at splitting time